### ISSUE TITLE
Allow rule plugins to declare they only allow one instance

### ISSUE THREAD

**Issue by daledavies** [daledavies/moodle-tool_registrationrules#35]
I think this makes sense. For example would we want more than one instance of hcaptcha on the registration form?

**Comment by daledavies** [comment]
I have a branch with a fix for this locally, not going to make a PR yet as it will require PR #55 to be merged first.

**Comment by daledavies** [comment]
Maybe, actually, single instances should be the default and plugins should declare they can use multiple instances.

**Comment by phager-at** [comment]
I agree with that idea. Most plugins would only need a single instance, so that'd be a sensible default.

**Comment by daledavies** [comment]
Implemented in https://github.com/daledavies/moodle-tool_registrationrules/commit/2bdf70ece8bbc52c308b76ffbe7baa8e39ba2531

This defaults to just a single instance unless a rule plugin implements the `multiple_instances` interface, which will allow multiple instances of that plugin to be created.
